tcms/testruns/tests/test_permissions.py

```diff
@@ -128,3 +128,31 @@ def verify_post_with_permission(self):
             self.assertEqual(case.history.latest().history_id, case_run.case_text_version)
             self.assertEqual(last_run.build, case_run.build)
             self.assertEqual(None, case_run.close_date)
+
+
+class GetTestRunViewTestCase(PermissionsTestCase):
+    permission_label = 'django_comments.add_comment'
+    http_method_names = ['get']
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.product = factories.ProductFactory(name='Kiwi')
+        cls.build = factories.BuildFactory(product=cls.product)
+        cls.test_run = factories.TestRunFactory(
+                                      build=cls.build)
```

Comment:
Technically you just need the URL below for which you only need test_run, which doesn't need the rest of the attributes (will be auto created).